src/main/java/io/github/genomicdatainfrastructure/daam/mappers/ApplicationNotInCorrectStateExceptionMapper.java

```diff
@@ -0,0 +1,14 @@
+package io.github.genomicdatainfrastructure.daam.mappers;
+
+import io.github.genomicdatainfrastructure.daam.exceptions.ApplicationNotInCorrectStateException;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.ExceptionMapper;
+import jakarta.ws.rs.ext.Provider;
+
+@Provider
+public class ApplicationNotInCorrectStateExceptionMapper implements ExceptionMapper<ApplicationNotInCorrectStateException> {
+    @Override
+    public Response toResponse(ApplicationNotInCorrectStateException exception) {
+        return Response.status(Response.Status.BAD_REQUEST).entity(exception.getMessage()).build();
```

Comment:
important: `BAD_REQUEST` is misleading here, it should be either `CONFLICT - 409`, `UNPROCESSABLE_ENTITY - 422` or `PRECONDITION_REQUIRED - 428`. Perhaps the last one fits better.